### PR TITLE
Bump Gateway API to v0.8.0-rc2

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Updates https://github.com/projectcontour/contour/issues/5561

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Initially bumps to main ahead of the v0.8.0 release

**Comment by skriss** [comment]
:+1: conformance much faster not waiting for terminating pods to go away!

File: examples/gateway/01-admission_webhook.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ spec:
       containers:
       - name: webhook
         image: registry.k8s.io/gateway-api/admission-server:v0.7.1
```
Comment (skriss) [inline examples/gateway/01-admission_webhook.yaml:59]: I'd expect this to be tagged `v0.8.0-rc.1`, raised in slack to Rob.

**Comment by sunjayBhatia** [comment]
mirror test still flaky

**Review by skriss — COMMENTED** [review]
updates all LGTM 💯

File: test/scripts/run-gateway-conformance.sh [hunk 2]
```diff
@@ -58,9 +58,9 @@ else
...
   cd gateway-api
   git checkout "${GATEWAY_API_VERSION}"
-  # TODO: Keep the list of skipped features in sync with
+  # Keep the list of skipped features in sync with
   # test/conformance/gatewayapi/gateway_conformance_test.go.
-  # Can implement with the -skip flag available with go 1.20
-  # or if Gateway API supports skipping tests via custom flag.
-  go test -timeout=40m ./conformance -gateway-class=contour -all-features -exempt-features=Mesh
+  go test -timeout=40m ./conformance -gateway-class=contour -all-features \
+    -exempt-features=Mesh,SupportGatewayClassRoutability,SupportGatewayPublicRoutability,SupportGatewayPrivateRoutability,SupportGatewayClusterRoutability \
```
Comment (dprotaso) [inline test/scripts/run-gateway-conformance.sh:64]: nit: the routability tests were removed

Comment (sunjayBhatia) [reply test/scripts/run-gateway-conformance.sh]: good catch 👍🏽

**Review by skriss — APPROVED** [review]
LGTM, as discussed may as well merge now to help with CI and we'll upgrade to the GA release when it's out

**Comment by sunjayBhatia** [comment]
Will move over the changelog etc. as needed when we bump to the official release

**Pull request merged**